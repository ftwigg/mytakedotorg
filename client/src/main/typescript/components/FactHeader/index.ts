import FactHeader from "./FactHeader";

export default FactHeader;
