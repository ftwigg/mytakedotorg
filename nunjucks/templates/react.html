{% extends "reactCore.html" %}

{% block bundle %}
	<!-- Main -->
	<script src="/app.bundle.js"></script>
{% endblock %}
