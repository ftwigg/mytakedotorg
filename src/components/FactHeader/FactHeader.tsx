import * as React from "react";
import * as ReactDOM from "react-dom";
import { FoundationNode } from "../Foundation";
import { getNodeArray } from "../../utils/functions";

interface FactHeaderProps {
  heading: string;
  className: string;
  onClearClick: () => void;
  onScroll: (headerHidden: boolean) => void;
  textIsHighlighted: boolean;
}

interface FactHeaderState {}

class FactHeader extends React.Component<
FactHeaderProps,
  FactHeaderState
> {
  private header: HTMLDivElement;
  static headerHeight = 219;
  constructor(props: FactHeaderProps) {
    super(props);
  }
  handleClearClick = () => {
    this.props.onClearClick();
  };
  handleScroll = () => {
    this.props.onScroll(this.header.getBoundingClientRect().top <= 0);
  };
  componentDidMount() {
    window.addEventListener("scroll", this.handleScroll);
  }
  componentWillUnmount() {
    window.removeEventListener("scroll", this.handleScroll);
  }
  render() {
    const headerContent = (
      <div>
        <h2 className={"document__heading"}>
          {this.props.heading}
        </h2>
        {this.props.textIsHighlighted
          ? <button
              className="document__button"
              onClick={this.handleClearClick}
            >
              Clear Selection
            </button>
          : null}
      </div>
    );

    return (
      <div ref={(header: HTMLDivElement) => (this.header = header)}>
        <div className={this.props.className}>
          {headerContent}
        </div>
        <div className={"document__header document__header--fixed"}>
          {headerContent}
        </div>
      </div>
    );
  }
}

export default FactHeader;
