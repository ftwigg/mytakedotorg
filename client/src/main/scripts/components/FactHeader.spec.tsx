import * as React from "react";
import * as renderer from "react-test-renderer";
import {} from "jest";
import FactHeader from "./FactHeader";

const documentProps = {
  heading: "Bill of Rights",
  isDocument: true,
  onClearClick: jest.fn(),
  onSetClick: jest.fn(),
  onScroll: jest.fn()
};

test("Fixed with text highlighted", () => {
  const tree = renderer
    .create(
      <FactHeader {...documentProps} isFixed={true} textIsHighlighted={true} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("Fixed without text highlighted", () => {
  const tree = renderer
    .create(
      <FactHeader {...documentProps} isFixed={true} textIsHighlighted={false} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("Scrolling with text highlighted", () => {
  const tree = renderer
    .create(
      <FactHeader {...documentProps} isFixed={false} textIsHighlighted={true} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("Scrolling without text highlighted", () => {
  const tree = renderer
    .create(
      <FactHeader {...documentProps} isFixed={false} textIsHighlighted={false} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
