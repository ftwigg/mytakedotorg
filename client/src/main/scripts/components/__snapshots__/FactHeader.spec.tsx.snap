// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fixed with text highlighted 1`] = `
<div>
  <div
    className="document__header document__header--hidden"
  >
    <div>
      <h2
        className="document__heading"
      >
        Bill of Rights
      </h2>
      <div
        className="document__header-actions"
      >
        <button
          className="document__button document__button--blue"
          onClick={[Function]}
        >
          Clear Selection
        </button>
        <button
          className="document__button document__button--red"
          onClick={[Function]}
        >
          Give your Take on this
        </button>
      </div>
    </div>
  </div>
  <div
    className="document__header document__header--fixed"
  >
    <div>
      <h2
        className="document__heading"
      >
        Bill of Rights
      </h2>
      <div
        className="document__header-actions"
      >
        <button
          className="document__button document__button--blue"
          onClick={[Function]}
        >
          Clear Selection
        </button>
        <button
          className="document__button document__button--red"
          onClick={[Function]}
        >
          Give your Take on this
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Fixed without text highlighted 1`] = `
<div>
  <div
    className="document__header document__header--hidden"
  >
    <div>
      <h2
        className="document__heading"
      >
        Bill of Rights
      </h2>
      <div
        className="document__header-actions"
      >
        <p
          className="document__instructions"
        >
          Highlight something to give your Take
        </p>
      </div>
    </div>
  </div>
  <div
    className="document__header document__header--fixed"
  >
    <div>
      <h2
        className="document__heading"
      >
        Bill of Rights
      </h2>
      <div
        className="document__header-actions"
      >
        <p
          className="document__instructions"
        >
          Highlight something to give your Take
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Scrolling with text highlighted 1`] = `
<div>
  <div
    className="document__header document__header--visible"
  >
    <div>
      <h2
        className="document__heading"
      >
        Bill of Rights
      </h2>
      <div
        className="document__header-actions"
      >
        <button
          className="document__button document__button--blue"
          onClick={[Function]}
        >
          Clear Selection
        </button>
        <button
          className="document__button document__button--red"
          onClick={[Function]}
        >
          Give your Take on this
        </button>
      </div>
    </div>
  </div>
  <div
    className="document__header document__header--hidden"
  >
    <div>
      <h2
        className="document__heading"
      >
        Bill of Rights
      </h2>
      <div
        className="document__header-actions"
      >
        <button
          className="document__button document__button--blue"
          onClick={[Function]}
        >
          Clear Selection
        </button>
        <button
          className="document__button document__button--red"
          onClick={[Function]}
        >
          Give your Take on this
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Scrolling without text highlighted 1`] = `
<div>
  <div
    className="document__header document__header--visible"
  >
    <div>
      <h2
        className="document__heading"
      >
        Bill of Rights
      </h2>
      <div
        className="document__header-actions"
      >
        <p
          className="document__instructions"
        >
          Highlight something to give your Take
        </p>
      </div>
    </div>
  </div>
  <div
    className="document__header document__header--hidden"
  >
    <div>
      <h2
        className="document__heading"
      >
        Bill of Rights
      </h2>
      <div
        className="document__header-actions"
      >
        <p
          className="document__instructions"
        >
          Highlight something to give your Take
        </p>
      </div>
    </div>
  </div>
</div>
`;
